skynet/modules/ttt/rag/stores/faiss.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import shutil
+import time
+
+from uuid import uuid4
+
+import faiss
+
+from langchain_community.docstore.in_memory import InMemoryDocstore
+from langchain_community.vectorstores import FAISS
+from langchain_community.vectorstores.utils import DistanceStrategy
+
+from skynet.env import use_s3, vector_store_path
+from skynet.logs import get_logger
+from skynet.modules.ttt.rag.vector_store import SkynetVectorStore
+
+log = get_logger(__name__)
+
+
+class FAISSVectorStore(SkynetVectorStore):
+    def __init__(self):
+        super().__init__()
+
+        if use_s3:
+            from skynet.modules.ttt.rag.stores.s3 import RagS3
+
+            log.info('Using S3 for vector store')
+            self.s3 = RagS3()
+
+    def get_vector_store_path(self, store_id):
+        return f'{vector_store_path}/faiss/{store_id}'
+
+    async def initialize(self):
+        await super().initialize()
+
+        if use_s3:
+            await self.s3.replicate(self.get_vector_store_path)
+
+    async def get(self, store_id):
+        try:
+            return FAISS.load_local(
+                self.get_vector_store_path(store_id), self.embedding, allow_dangerous_deserialization=True
+            )
+        except RuntimeError:
+            log.info(f'Vector store with id {store_id} not found')
+            return None
```

Comment:
Whay is the caller supposed to do if they get None?

Reply:
that would be the `else` case I was talking about for `if vector_store:` in `processor`